### PR TITLE
fixed background / overlay image not recalculating zoom value

### LINKED ISSUE #5280: SVG backgroundImage does not re-render on canvas zoom
## Version
2.4.0

## Information about environment
All browser

## Steps to reproduce
Set an svg as the canvas background image, then use canvas.zoomToPoint to zoom in.

## Expected Behavior
SVG re-render to zoom factor and look sharp. was working with version 2.2.
http://jsfiddle.net/at60ojdc/

## Actual Behavior
SVG does not re-render so it get zoom but look super pixelated.
http://jsfiddle.net/t34La5yk/

### PR DESCRIPTION
**Pull request by asturur**
Add a canvas reference so background or overlay image know the zoom value

Close #5280

### CODE REVIEW

File: src/static_canvas.class.js [hunk 1]
```diff
@@ -459,13 +467,18 @@
     __setBgOverlayImage: function(property, image, callback, options) {
       if (typeof image === 'string') {
         fabric.util.loadImage(image, function(img) {
-          img && (this[property] = new fabric.Image(img, options));
+          if (img) {
+            var instance = new fabric.Image(img, options);
+            this[property] = instance;
+            instance.canvas = this;
+          }
           callback && callback(img);
```
Comment (durga598) [inline src/static_canvas.class.js]: Can we return fabric image object i.e. instance, instead of img?

Comment (asturur) [reply src/static_canvas.class.js]: That is another method, is called `fabric.Image.fromUrl(...)`

Comment (durga598) [reply src/static_canvas.class.js]: Yes but on callback of `setBackgroundImage/ setOverlayImage` it returns image element not fabric.Image instance.

Comment (asturur) [reply src/static_canvas.class.js]: Does it even make sense that those callback return something?
You are right they should at least return the same thing.

I would remove the method completely to be honest. Devs should create theyr background object and set it normally.

**Pull request merged**